src/ast.js

```diff
@@ -148,6 +149,20 @@ export class ValueConverter extends Expression {
     while (i--) {
       expressions[i].connect(binding, scope);
     }
+    let converter = binding.lookupFunctions.valueConverters(this.name);
+    if (!converter) {
+      throw new Error(`No ValueConverter named "${this.name}" was found!`);
+    }
+    let signals = converter.signals;
+    if (signals === undefined) {
+      return;
+    }
+    // support both input type 'signal' & ['signal-1', 'signal-2']
+    signals = Array.isArray(signals) ? signals : [signals];
```

Comment:
This is a really hot path. Let's remove this line entirely to ensure the connect overhead remains minimal.

I don't think supporting string & array is necessary, just more to document & maintain. If we were to do that here's a good spot to perform the normalization, but again, I would prefer to keep the API surface small and strict:

https://github.com/aurelia/binding/blob/b6565eb116c212188feec296ae4782c3c1d95c89/src/value-converter-resource.js#L15-L17

Reply:
@jdanyow All done as suggested. Learnt to be more careful with `connect`. About the API, 99% of usage will be a single signal, I do think we need to support normalisation, but I'll let you make the call after this got merged. What do you think ?